modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/A11yPanel.tsx

```diff
@@ -0,0 +1,148 @@
+ * details.
+ */
+
+import React, {useState} from 'react';
+
+import Occurrence from './Occurrence';
+import Violation from './Violation';
+import Violations from './Violations';
+import {FilteredViolationsContextProvider} from './useFilteredViolations';
+
+import type {ImpactValue, Result} from 'axe-core';
+
+function isBetweenRange(childrenCount: number, newIndex: number) {
+	return 0 <= newIndex && newIndex <= childrenCount;
+}
+
+type SidebarPanelNavigatorProps = {
+	children: React.ReactNode;
+};
+
+type TNavigationState = {
+	occurrenceIndex?: number;
+	occurrenceName?: string;
+	violationsIndex: number;
+};
+
+enum PagesEnum {
+	Occurrence = 2,
+	Violation = 1,
+	Violations = 0,
+}
+
+function SidebarPanelsNavigator({children}: SidebarPanelNavigatorProps) {
+	const [activePageIndex, setActivePageIndex] = useState(
+		PagesEnum.Violations
+	);
+
+	const [navigationState, setNavigationState] = useState<TNavigationState>({
+		violationsIndex: 0,
+	});
+
+	return (
+		<div className="page-accessibility-tool__sidebar sidebar sidebar-light">
+			{React.Children.map(children, (child, index) => {
+				const childrenCount = React.Children.count(children);
+
+				if (index === activePageIndex && React.isValidElement(child)) {
+					return (
+						child &&
+						React.cloneElement(child, {
+							index: activePageIndex,
+							key: index,
+							navigationState,
+							next: (newPayload: TNavigationState) => {
+								const newIndex = activePageIndex + 1;
+
+								if (isBetweenRange(childrenCount, newIndex)) {
+									setActivePageIndex(newIndex);
+									setNavigationState(newPayload);
+								}
+							},
+							previous: (newPayload: TNavigationState) => {
+								const newIndex = activePageIndex - 1;
+
+								if (isBetweenRange(childrenCount, newIndex)) {
+									setActivePageIndex(newIndex);
+
+									if (newPayload) {
+										setNavigationState(newPayload);
+									}
+								}
+							},
+							...child.props,
+						})
+					);
+				}
+			})}
+		</div>
+	);
+}
+
+const IMPACT_PRIORITY = {
+	critical: 4,
+	minor: 1,
+	moderate: 2,
+	serious: 3,
+} as const;
+
+function getImpactPriority(impact: ImpactValue) {
+	if (impact) {
+		return IMPACT_PRIORITY[impact];
+	}
+
+	return IMPACT_PRIORITY.minor;
+}
+
+function sortByImpact(violations: Array<Result>) {
+	return violations.sort((currentViolation, nextViolation) => {
+		if (nextViolation.impact && currentViolation.impact) {
+			return (
+				getImpactPriority(nextViolation.impact) -
+				getImpactPriority(currentViolation.impact)
+			);
+		}
+
+		return 0;
+	});
+}
+
+type A11yPanelProps = {
+	violations: Array<Result>;
+};
+
+export function A11yPanel({violations}: A11yPanelProps) {
+	const sortedByImpactViolations = sortByImpact(violations);
+
+	return (
+		<FilteredViolationsContextProvider
+			value={{
+				filteredViolations: sortedByImpactViolations,
+				violations: sortedByImpactViolations,
```

Comment:
This is a little strange why we have two properties with the same values? in theory, just having violations should be enough.

Reply:
Removed! Just using `violations` as the value!